changelog/unreleased/issue-4529

```diff
@@ -0,0 +1,18 @@
+Enhancement: Add extra verification of data integrity before upload
+
+Hardware issues, or a bug in restic or its dependencies, could previously cause
+corruption in the files restic created and stored in the repository. Detecting
+such corruption previously required explicitly running the `check --read-data`
+or `check --read-data-subset` commands.
+
+To further ensure data integrity, even in the case of hardware or software bugs,
```

Comment:
Suggest changing "hardware or software bugs" to "hardware issues or software bugs", as issues and bugs are different things.

Reply:
oh indeed ^^ . Done.